src/gui/loglistwidget.cpp

```diff
@@ -82,9 +87,9 @@ void LogListWidget::appendLine(const QString &line, const Log::MsgType &type)
     auto *lbl = new QLabel(line);
     lbl->setTextFormat(Qt::RichText);
     lbl->setContentsMargins(4, 2, 4, 2);
+    m_maxRowHeight = std::max(m_maxRowHeight, lbl->height());
```

Comment:
Wouldn't this work? `item->setSizeHint(lbl->height());`

Reply:
![image](https://user-images.githubusercontent.com/34717789/77986808-a6f8b700-7335-11ea-86a0-65f8d27dc543.png)

Reply:
![image](https://user-images.githubusercontent.com/34717789/77986924-fccd5f00-7335-11ea-9f3f-a4d71be52ec0.png)
without custom theme

Reply:
OK. Another question, what if the log message contains a newline character? Your code will change every row to the height of two lines.

Reply:
storing each line-height will be overkill, do we really support such scenarios?

Reply:
have you looked at `lbl->height()` value? please share what you see.

Reply:
Seems it's incorrect to rely on widget size before it is shown.

Reply:
what about using [QFontMetrics](https://doc.qt.io/qt-5/qfontmetrics.html) to determine text height?

Reply:
the problem is stylesheet is set after the sizeHint Call which changed sizeHint but `widgetitem` doesn't know about changed `sizehint` thus the problem, this can be solved using event filter but   IDK if that would be an acceptable though (I feel like glassez will not like such solution)

Reply:
>I feel like glassez will not like such solution

I can't say anything useful since your allusion to "some solution using event filter" can really mean anything, either a valid solution or a dirty workaround... Please explain exactly what you want to do.

Reply:
please ignore it.

I think the only reason we require label inside `QListWidgetItem` is to do separate formatting for time and log message (time is grey while log follows other colors), #12144 already removes this behavior, so how about until that is merged we can remove this behavior here as well directly set  display and foreground roles of `QListWidgetItem`

Reply:
> I think the only reason we require label inside `QListWidgetItem` is to do separate formatting for time and log message (time is grey while log follows other colors), #12144 already removes this behavior, so how about until that is merged we remove this behavior here as well directly set display and foreground roles of `QListWidgetItem` that will fix the problem

@glassez @Chocobo1 would really like your opinion on this.

Reply:
>time is grey while log follows other colors

I think this is a very familiar (and apparently convenient) look-and-feel, so removing it can be very much disputable even in #12144.

Reply:
As an alternative it can be implemented as table-like view to provide different columns for time and message (and possible for log level).

Reply:
First will try using qstyleitemdelegate